mkdocs/docs/api.md

```diff
@@ -1533,3 +1533,111 @@ df.show(2)
 
 (Showing first 2 rows)
 ```
+
+### Polars
+
+PyIceberg interfaces closely with Polars Dataframes and LazyFrame which provides a full lazily optimized query engine interface on top of PyIceberg tables.
+
+<!-- prettier-ignore-start -->
+
+!!! note "Requirements"
+    This requires [`polars` to be installed](index.md).
```

Comment:
i know the other sections also do the same, but we should mention `pyiceberg['polars']` as a way to install this dep